### PR TITLE
conditionally disable bottleneck

### LINKED ISSUE #5424: Bottleneck bug with unusual strides - causes segfault or wrong number
```python
import numpy as np
import xarray as xr
data = np.zeros((1,500, 2)).transpose(1,2,0)
xarr = xr.DataArray(data, coords=[('A', range(500)), ('B', [0,1]), ('C', [0])])
xarr.max()                                  
```
The above either returns a very large non-zero number or segfaults. Due to https://github.com/pydata/bottleneck/issues/381.

Dual posting here in case this isn't able to get quickly fixed in bottleneck, as this is a pretty severe bug - especially on the occaions it returns the wrong number rather than segfaulting.

**Environment**:
```
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.7 (default, Mar 26 2020, 15:48:22) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 3.10.0-1160.21.1.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8
libhdf5: 1.10.4
libnetcdf: None

xarray: 0.14.1
pandas: 0.25.0
numpy: 1.16.6
scipy: 1.4.1
netCDF4: None
pydap: None
h5netcdf: None
h5py: 2.10.0
Nio: None
zarr: None
cftime: None
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: 1.3.2
dask: 2.10.1
distributed: 2.10.0
matplotlib: 3.1.3
cartopy: None
seaborn: 0.10.0
numbagg: installed
setuptools: 46.1.3.post20200330
pip: 20.0.2
conda: None
pytest: 5.2.4
IPython: 7.20.0
sphinx: 2.4.4
```

### PR DESCRIPTION
**Pull request by keewis**
As this came up in #5424 (and because I can't seem to reliably reproduce expected values in #4972 if it is enabled) this adds a option to disable `bottleneck`, even if it is installed.

In #5424 it was suggested to also allow replacing `bottleneck` with `numbagg`. If that's something we want (and `numbagg` supports the operations we need) I can try looking into renaming the new option to something like `xr.set_options(accelerate_with="bottleneck")` (or something else, if someone has a great idea).

Tests are missing because I have no idea how to check that this works (except by mocking bottleneck).

- [x] Closes #5424
- [x] Tests added
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by dcherian** [comment]
We use it in other places too (for e.g.): https://github.com/pydata/xarray/blob/c472f8a4c79f872edb9dcd7825f786ecb9aff5c0/xarray/core/nputils.py#L139-L147

> xr.set_options(accelerate_with="bottleneck")

I like this idea.

**Comment by max-sixty** [comment]
Nice!

pandas [uses](https://pandas.pydata.org/docs/reference/api/pandas.set_option.html) `use_bottleneck` — and same for numba / numexpr.

To what extent would bottleneck & numba be mutually exclusive? Until everything is implemented in numbagg, I guess they won't be, and we might want separate options. Having a single option would be nicer otherwise.

**Comment by keewis** [comment]
as discussed in the meeting, we will keep `use_bottleneck` for now. I still don't know how to test this, but otherwise this should be ready for reviews and merging.

**Review by dcherian — COMMENTED** [review]
LGTM. I guess you would have to mock bottleneck to properly test this.

**Comment by dcherian** [comment]
> so disabling bottleneck would fail the function. Should we just silently use bottleneck, or raise an error?

Ah this is a good test!

``` python
with xr.set_options(use_bottleneck=False):
	with pytest.raises(...):
		dataarray.ffill()
```

IMO it should raise an error so that `use_bottleneck` is a "global" control on whether xarray uses bottleneck or not. The context manager gives the user some flexibility to opt-in to using bottleneck where they want to.

**Comment by keewis** [comment]
I'm not sure how to test rolling: monkeypatching `bottleneck.move_sum` does not work because `Rolling` only accesses that on import, i.e. before the test is executed.

Everything else is done, though.

File: xarray/core/dataset.py [hunk 1]
```diff
@@ -6228,6 +6228,9 @@ def rank(self, dim, pct=False, keep_attrs=None):
         ranked : Dataset
             Variables that do not depend on `dim` are dropped.
         """
+        if not OPTIONS["use_bottleneck"]:
+            raise RuntimeError("rank requires bottleneck to be enabled")
```
Comment (dcherian) [inline xarray/core/dataset.py:6232]:
```suggestion
            raise RuntimeError("rank requires bottleneck to be enabled. Please run xr.set_options(use_bottleneck=True)")
```

**Review by dcherian — APPROVED** [review]
just a suggestion on the error message.

Can you open a new issue about testing for rolling? I think it is OK to merge without that.

**Comment by dcherian** [comment]
Thanks @keewis. We can extend the tests for rolling later.

**Pull request merged**